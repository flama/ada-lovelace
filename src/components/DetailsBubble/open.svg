<?xml version="1.0" encoding="UTF-8"?>
<svg width="8px" height="7px" viewBox="0 0 8 7" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 42 (36781) - http://www.bohemiancoding.com/sketch -->
    <title>Page 1</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Interações-Gráfico-@2x" transform="translate(-756.000000, -3349.000000)">
            <g id="Group-5" transform="translate(706.000000, 3342.000000)">
                <g id="Page-1" transform="translate(50.000000, 6.500000)">
                    <polygon id="Fill-1" points="0 8 8 8 8 0 0 0"></polygon>
                    <polygon id="Fill-2" fill="#FFA9C7" points="1.33333333 1.66666667 0.666666667 1.66666667 0.666666667 7.33333333 6.33333333 7.33333333 6.33333333 6.66666667 1.33333333 6.66666667"></polygon>
                    <polygon id="Fill-3" fill="#FFA9C7" points="2.66666667 0.666666667 2.66666667 1.33333333 6.19533333 1.33333333 2.43066667 5.09733333 2.90266667 5.56933333 6.66666667 1.80466667 6.66666667 5.33333333 7.33333333 5.33333333 7.33333333 0.666666667"></polygon>
                </g>
            </g>
        </g>
    </g>
</svg>