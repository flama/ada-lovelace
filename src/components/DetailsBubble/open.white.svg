<?xml version="1.0" encoding="UTF-8"?>
<svg width="8px" height="7px" viewBox="0 0 8 7" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 43.2 (39069) - http://www.bohemiancoding.com/sketch -->
    <title>Page 1</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Interações-Gráfico-3-@2x" transform="translate(-1123.000000, -855.000000)">
            <g id="Group-8" transform="translate(133.000000, 304.000000)">
                <g id="Page-1" transform="translate(990.000000, 550.500000)">
                    <polygon id="Fill-1" points="0 8 8 8 8 0 0 0"></polygon>
                    <polygon id="Fill-2" fill="#FFFFFF" points="1.33333333 1.66666667 0.666666667 1.66666667 0.666666667 7.33333333 6.33333333 7.33333333 6.33333333 6.66666667 1.33333333 6.66666667"></polygon>
                    <polygon id="Fill-3" fill="#FFFFFF" points="2.66666667 0.666666667 2.66666667 1.33333333 6.19533333 1.33333333 2.43066667 5.09733333 2.90266667 5.56933333 6.66666667 1.80466667 6.66666667 5.33333333 7.33333333 5.33333333 7.33333333 0.666666667"></polygon>
                </g>
            </g>
        </g>
    </g>
</svg>