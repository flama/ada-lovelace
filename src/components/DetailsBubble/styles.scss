@import '../../config';

#details-bubble {
  width: 0;
  height: 0;
  overflow: hidden;
  border-radius: 50%;
  background-color: $selected;
  position: absolute;

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;

  margin-top: 0;
  margin-left: 0;
  transition: 0.5s ease-in-out;
  transition-property: height, width, margin-top, margin-left;

  &.show {
    width: 182px;
    height: 182px;
    margin-top: -91px;
    margin-left: -91px;
    transition: 0.5s ease-in-out;
    transition-property: height, width, margin-top, margin-left;
  }

  .content {
    flex-grow: 1;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    > * {
      max-width: $space-xxxl;
      text-align: center;
    }

    .name {
      font-size: $font-size-deca;
      line-height: $line-height-octo;
      font-weight: $regular;
    }
  }

  .description, a { font-size: $font-size-centi; }

  .description {
    line-height: $line-height-mili;
    color: $feminist-purple;
  }

  a {
    color: $color-ghost;
    margin-top: $space-xs;
    margin-bottom: $space-xs;
    flex-grow: 0.25;

    // TODO: make svg be a thing
    // &.external:after {
    //   height: 1em;
    //   width: 1em;
    //   margin-left: $space-xxs;
    //   margin-right: $space-xxs;
    //   display: inline-block;
    //   content: "";
    //   background-image: url('./open.svg');
    //   background-repeat: no-repeat;
    //   background-size: 100%;
    //   background-position: center center;
    // }
  }
}
