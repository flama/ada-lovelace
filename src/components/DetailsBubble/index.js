import React from 'react'
import PropTypes from 'prop-types'

import './styles.scss'

const DetailsBubble = props => (
  <div id="details-bubble">
  <a onClick={ () => props.close() } className="close">Close</a>
  <div className="content">
  <h3 className="name"></h3>
  <p className="description"></p>
  </div>
  <a className="external" target="_blank">More info</a>
  </div>
)

DetailsBubble.propTypes = {
  close: PropTypes.func
}

export default DetailsBubble
