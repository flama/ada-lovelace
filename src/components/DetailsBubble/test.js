import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import DetailsBubble from '../DetailsBubble'

// TODO: test `close` functionality
describe('DetailsBubble', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DetailsBubble />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
