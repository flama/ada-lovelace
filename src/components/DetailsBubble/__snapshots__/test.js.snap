// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsBubble renders correctly 1`] = `
<div
  id="details-bubble"
>
  <a
    className="close"
    onClick={[Function]}
  >
     Close 
  </a>
  <div
    className="content"
  >
    <h3
      className="name"
    />
    <p
      className="description"
    />
  </div>
  <a
    className="external"
    target="_blank"
  >
     More info 
  </a>
</div>
`;
